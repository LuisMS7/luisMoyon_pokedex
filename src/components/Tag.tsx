import React from "react";
import { StyleSheet, Text, View } from "react-native";
import { BOLD_WEIGHT, TEXT_SMALL_SIZE } from "@lib/constants";
import { TagType } from "@src/types/tag.type";

type TagProps = TagType;

const Tag = ({
	text,
	backgroundColor,
	textColor = "white",
	borderColor = "transparent",
	fontWeight = BOLD_WEIGHT,
}: TagProps) => {
	return (
		<View
			style={{
				...styles.tag,
				backgroundColor: backgroundColor,
				borderColor: borderColor,
			}}
		>
			<Text
				style={{
					...styles.tagText,
					color: textColor,
					fontWeight: fontWeight,
				}}
			>
				{text}
			</Text>
		</View>
	);
};

const styles = StyleSheet.create({
	tag: {
		borderRadius: 1000,
		alignItems: "center",
		justifyContent: "center",
		paddingHorizontal: 10,
		paddingVertical: 5,
		borderWidth: 1,
	},
	tagText: {
		color: "white",
		fontWeight: BOLD_WEIGHT,
		fontSize: TEXT_SMALL_SIZE,
	},
});

export default Tag;
