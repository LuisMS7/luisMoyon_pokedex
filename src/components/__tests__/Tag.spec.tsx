import { TagType } from "@src/types/tag.type";
import Tag from "@src/components/Tag";
import { render, screen } from "@testing-library/react-native";
import React from "react";

describe("Tag", () => {
	it.each([
		[{ text: "tag1", backgroundColor: "red" }],
		[
			{
				text: "tag2",
				backgroundColor: "red",
				textColor: "black",
				borderColor: "blue",
			},
		],
		[
			{
				text: "tag3",
				backgroundColor: "transparent",
				textColor: "black",
				borderColor: "green",
			},
		],
	])("should render correctly '%s' tag", (tag: TagType) => {
		render(<Tag {...tag} />);
		expect(screen.toJSON()).toMatchSnapshot();
	});
});
