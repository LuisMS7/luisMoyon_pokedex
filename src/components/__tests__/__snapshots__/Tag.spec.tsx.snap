// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tag should render correctly '{
  text: 'tag2',
  backgroundColor: 'red',
  textColor: 'black',
  borderColor: 'blue'
}' tag 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "red",
      "borderColor": "blue",
      "borderRadius": 1000,
      "borderWidth": 1,
      "justifyContent": "center",
      "paddingHorizontal": 10,
      "paddingVertical": 5,
    }
  }
>
  <Text
    style={
      {
        "color": "black",
        "fontSize": 17.89194962785868,
        "fontWeight": "800",
      }
    }
  >
    tag2
  </Text>
</View>
`;

exports[`Tag should render correctly '{
  text: 'tag3',
  backgroundColor: 'transparent',
  textColor: 'black',
  borderColor: 'green'
}' tag 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "transparent",
      "borderColor": "green",
      "borderRadius": 1000,
      "borderWidth": 1,
      "justifyContent": "center",
      "paddingHorizontal": 10,
      "paddingVertical": 5,
    }
  }
>
  <Text
    style={
      {
        "color": "black",
        "fontSize": 17.89194962785868,
        "fontWeight": "800",
      }
    }
  >
    tag3
  </Text>
</View>
`;

exports[`Tag should render correctly '{ text: 'tag1', backgroundColor: 'red' }' tag 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "red",
      "borderColor": "transparent",
      "borderRadius": 1000,
      "borderWidth": 1,
      "justifyContent": "center",
      "paddingHorizontal": 10,
      "paddingVertical": 5,
    }
  }
>
  <Text
    style={
      {
        "color": "white",
        "fontSize": 17.89194962785868,
        "fontWeight": "800",
      }
    }
  >
    tag1
  </Text>
</View>
`;
